src/config/config.rs

```diff
@@ -40,19 +40,49 @@ pub struct Config {
 #[derive(Debug, Parser)]
 /// Storage config
 pub struct StorageConfig {
-    #[clap(long = "storage-type", value_name = "VALUE", default_value = "none")]
-    /// Storage type: S3,None
+    #[clap(long = "storage-type", value_name = "VALUE", default_value = "fs")]
+    /// Storage type: s3, fs
     pub storage_type: String,
-    #[clap(
-        long = "storage-cache-capacity",
-        value_name = "VALUE",
-        default_value_t = 1073741824
-    )]
-    /// Set memory cache capacity, default is 1GB
-    pub cache_capacity: usize,
+    #[clap(flatten)]
+    /// The memory cache config
+    pub memory_cache_config: MemoryCacheConfig,
     #[clap(flatten)]
     /// S3 storage config
     pub s3_storage_config: S3StorageConfig,
+    #[clap(long = "storage-fs-root", value_name = "VALUE", default_value_t)]
+    /// The root of FS backend
+    pub fs_storage_root: String,
+}
+
+/// Memory cache config
+#[derive(Debug, Parser)]
+pub struct MemoryCacheConfig {
+    /// The capacity of memory cache in blocks, default is 16384 blocks.
+    #[clap(
+        long = "storage-mem-cache-capacity",
+        value_name = "VALUE",
+        default_value_t = 16384
+    )]
+    pub capacity: usize,
+    /// The limitation of the message queue of the write back task, default is
+    /// 1000
+    #[clap(
+        long = "storage-mem-cache-command-limit",
+        value_name = "VALUE",
+        default_value_t = 1000
+    )]
+    pub command_queue_limit: usize,
+    /// A flag whether the cache runs in write-back policy, default is false
+    #[clap(long = "storage-mem-cache-write-back")]
+    pub write_back: bool,
+    /// A soft limit of evict policy for write back task.
+    ///
+    /// It's a fraction with the form of `a,b`, which means that
+    /// the soft limit is `a/b` of the capacity.
+    ///
+    /// **Note**: `b` cannot be set to 0.
+    #[clap(long = "storage-mem-cache-soft-limit", default_value = "3,5")]
```

Comment:
Using floating-point numbers here would provide a better experience for the user.

Reply:
I tried, but it's not easy. I can't cast a `usize` to `f64` safely, and of course it's not safe to cast `f64` to `usize`. If they are not in the same type, I can't multiply them together. Therefore I chose to use fraction to represent the soft limit.